polliwog/polyline/test_polyline_object.py

```diff
@@ -1113,3 +1113,53 @@ def test_section_edge_case():
     np.testing.assert_array_equal(broken[0].v, vs[:11])
     np.testing.assert_array_equal(broken[1].v, vs[10:12])
     np.testing.assert_array_equal(broken[2].v, vs[11:])
+
+
+def test_point_along_curve():
+    example_vs = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [1.0, 1.0, 0.0],
+            [1.0, 3.0, 0.0],
+            [2.0, 3.0, 0.0],
+            [2.0, 4.0, 0.0],
+        ]
+    )
+    polyline = Polyline(v=example_vs, is_closed=False)
+    np.testing.assert_array_almost_equal(
+        polyline.point_along_curve(0.2), np.array([1.0, 0.2, 0.0])
+    )
+    np.testing.assert_array_almost_equal(
+        polyline.point_along_curve(0.8), np.array([1.8, 3.0, 0.0])
+    )
+    np.testing.assert_array_almost_equal(
+        polyline.point_along_curve(0.05), np.array([0.3, 0.0, 0.0])
+    )
+    np.testing.assert_array_almost_equal(
+        polyline.point_along_curve(0.95), np.array([2.0, 3.7, 0.0])
+    )
+
+
+def test_point_along_curve_errors():
+    vs = np.arange(108).reshape(36, 3)
+    closed_polyline = Polyline(v=vs, is_closed=True)
+    open_polyline = Polyline(v=vs, is_closed=False)
+    with pytest.raises(ValueError, match="Must be an open polyline"):
+        closed_polyline.point_along_curve(0.5)
+    with pytest.raises(
+        ValueError, match="percentage_along_curve must be a floating point number"
+    ):
+        open_polyline.point_along_curve(2)
+    with pytest.raises(
+        ValueError, match="percentage_along_curve must be a value between 0 and 1"
+    ):
+        open_polyline.point_along_curve(2.5)
+    with pytest.raises(
+        ValueError, match="percentage_along_curve must be a value between 0 and 1"
+    ):
+        open_polyline.point_along_curve(0.0)
+    with pytest.raises(
+        ValueError, match="percentage_along_curve must be a value between 0 and 1"
+    ):
+        open_polyline.point_along_curve(1.0)
```

Comment:
Any reason not to support 0 and 1? I think it would be better to let them return the starting and endpoints.